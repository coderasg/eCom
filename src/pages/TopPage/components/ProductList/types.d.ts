declare module 'Products' {
  export type Product = {
    id: number;
    name: string;
    description: string;
    like_count: number;
    comment_count: number;
    price: number;
    is_sold_out: boolean;
    shipping_fee: string;
    image: string;
    category_id: number;
  };

  export type Products = {
    data: Array<Product>;
  };

  export type ProductCategories = {
    data: Array<{
      id: number;
      name: string;
    }>;
  };
}
