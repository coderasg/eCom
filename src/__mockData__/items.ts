export default {
  'data': [
    {
      'id': 1,
      'name': 'Light pink shoes',
      'description': 'Pre-owned, in good condition with signs of wear, may just need to be cleaned or washed.  Please see photos for style & condition.',
      'like_count': 91,
      'comment_count': 59,
      'price': 51,
      'is_sold_out': false,
      'shipping_fee': '送料込み',
      'image': 'images/image_1.png',
      'category_id': 2
    },
    {
      'id': 2,
      'name': 'Vegan leather bag',
      'description': 'Black tote in excellent condition.\n\nFabric:Vegan leather\n Measurement: 17"L 0.5" W 11.75"H\n Handles about 8.5',
      'like_count': 0,
      'comment_count': 0,
      'price': 10000,
      'is_sold_out': true,
      'shipping_fee': '送料込み',
      'image': 'images/image_2.png',
      'category_id': 2
    }
  ]
};
  