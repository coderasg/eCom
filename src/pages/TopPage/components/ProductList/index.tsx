import React from 'react';

import useProductList from '../../../../hooks/fetchAPIs/useProductList';

import ProductListItem from '../ProductListItem';

import styles from './productList.module.scss';

type Props = {
  searchStr: string;
  selectedId: number|null;
};

function ProductList(props: Props) {
  const { searchStr, selectedId } = props;
  const { data: products, error } = useProductList({
    category: selectedId ? `${selectedId}` : '',
    searchStr: searchStr || ''
  });

  if (error) {
    return (
      <div data-testid="ProductList-Error" className={styles.productListError}>
        <h1>{`failed to load: ${error}`}</h1>
      </div>
    );
  }

  if (!products) {
    return (
      <h1 data-testid="ProductList-Loading" className={styles.productListError}>{'Loadiding..............'}</h1>
    );
  }

  return (
    <div data-testid="ProductList-wrapper" className={styles.productList}>
      <section className={styles.listContainer} >
        {Array.isArray(products) && !products.length ? (
          <h1>Sorry No Data Found</h1>
        ) : (
          products.map((product) => (
            <ProductListItem key={product.id} data={product} />
          ))
        )}
      </section>
    </div>
  );
}

export default ProductList;
