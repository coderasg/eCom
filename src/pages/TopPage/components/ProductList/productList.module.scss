.productList {
  .listContainer {
    display: flex;
    flex-wrap: wrap;
    justify-content: space-between;
    padding-top: 20px;
  }
}

.productListError {
  min-height: 50vh;
}