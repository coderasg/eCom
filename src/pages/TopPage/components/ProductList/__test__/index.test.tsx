import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import useSWR from 'swr';

import items from './../../../../../__mockData__/items';

import ProductList from '../index';

jest.mock('swr');

(useSWR as jest.Mock).mockReturnValue({
  data: {
    data: [items.data[0], items.data[1]],
  },
});

describe('(Component) ProductList', () => {
  const defaultProps = {
    searchStr: '',
    selectedId: 2,
  };

  test('should pass snapshot test', () => {
    const wrapper = renderer.create(<BrowserRouter><ProductList {...defaultProps} /></BrowserRouter>);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
