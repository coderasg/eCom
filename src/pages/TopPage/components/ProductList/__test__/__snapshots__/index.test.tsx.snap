// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) ProductList should pass snapshot test 1`] = `
<div
  data-testid="ProductList-wrapper"
>
  <section>
    <div
      data-testid="ProductListItem-wrapper"
    >
      <a
        href="/1"
        onClick={[Function]}
      >
        <div
          className="undefined undefined"
        >
          <img
            data-testid="ProductImage-wrapper"
            src="images/image_1.png"
          />
        </div>
        <section>
          <div>
            Light pink shoes
          </div>
          <div>
            <span>
              Ұ51
            </span>
            <span
              data-testid="Like-wrapper"
            >
              <SvgIcon
                height={15}
                width={15}
              />
              <span>
                91
              </span>
            </span>
          </div>
        </section>
      </a>
    </div>
    <div
      data-testid="ProductListItem-wrapper"
    >
      <a
        href="/2"
        onClick={[Function]}
      >
        <div
          className="undefined undefined"
        >
          <div>
            <span>
              SOLD
            </span>
          </div>
          <img
            data-testid="ProductImage-wrapper"
            src="images/image_2.png"
          />
        </div>
        <section>
          <div>
            Vegan leather bag
          </div>
          <div>
            <span>
              Ұ10000
            </span>
            <span
              data-testid="Like-wrapper"
            >
              <SvgIcon
                height={15}
                width={15}
              />
              <span>
                0
              </span>
            </span>
          </div>
        </section>
      </a>
    </div>
  </section>
</div>
`;
